test/dotnet-run.UnitTests/GivenARunCommand.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using FluentAssertions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.TestFramework;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using NuGet.Frameworks;
+using Xunit;
+
+namespace Microsoft.DotNet.Tools.Run.Tests
+{
+    public class GivenARunCommand : TestBase
+    {
+        private const int RunExitCode = 29;
+
+        [Fact]
+        public void ItDoesntRedirectStandardOutAndError()
+        {
+            TestInstance instance = TestAssetsManager.CreateTestInstance("TestAppSimple")
+                                         .WithLockFiles();
+
+            new BuildCommand(instance.TestRoot)
+                .Execute()
+                .Should()
+                .Pass();
+
+            RunCommand runCommand = new RunCommand(new FailOnRedirectOutputCommandFactory());
+            runCommand.Project = instance.TestRoot;
+
+            runCommand.Start()
+                .Should()
+                .Be(RunExitCode);
+        }
+
+        private class FailOnRedirectOutputCommandFactory : ICommandFactory
```

Comment:
I've never really used Moq before, and if it makes it easier to do this, then I'll be glad to learn.